HeroGame/main.py

```diff
@@ -0,0 +1,83 @@
+import math
+
+from hero import Hero
+from monster import Monster
+
+
+def print_current_game_status(hero: Hero, monster: Monster) -> None:
+    print("_" * 10)
+    print("Hero and Monster stats:")
+    print(f"Hero level: {hero.level}, Hero Health: {hero.hp}, Hero Damage: {hero.damage}, Hero Coins: {hero.coins}"
+          f"\nMonster level: {monster.level}, Monster health: {monster.hp}, Monster Damage: {monster.damage}")
+
+    print(f"You need a total of {Hero.level_up_threshold * (hero.level+1)} coins to level up")
+    if hero.coins >= Hero.level_up_threshold * (hero.level+1):
+        print("You have enough coins to level up!")
+    print("_" * 10)
+
+def handle_user_choice(user_choice: str, hero: Hero, monster: Monster):
+    if user_choice == "1":
+        hero.defending = True
+        print("You are now defending")
+    elif user_choice == "2":
+        hero.attack(monster)
+        print("You attacked the monster")
+    elif user_choice == "3":
+        hero.heal()
+        print(f"Healed {math.ceil(hero.hp * Hero.heal_percent)} health")
+    elif user_choice == "4":
+        hero.level_up()
+    hero.increase_coins(1)
+
+
+def choose_action(hero_name: str):
+    print(f"Its your turn {hero_name}, what would you like to do now?")
+    return input("1 -> Defend\n2 -> Attack\n3 -> Heal\n4 -> Level Up\n")
+
+
+def handle_monsters_turn(hero, monster):
+    monster.attack(hero)
```

Comment:
this does not need to be in a functions, it is a action of one line, and very detailed move from the name of the object and function